content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -0,0 +1,456 @@
+ 
+     System.out.println("Sleeping");
+ 
+     return true;
+ 
+ }
+ 
+ public boolean isEating(){
+ 
+     System.out.println("Eating");
+ 
+     return true;
+ 
+ }
+ 
+ public boolean isbarking(){
+ 
+     System.out.println("Dog barking...");
+ 
+     return true;
+ 
+ }
+ 
+ public boolean isRunning(){
+ 
+     System.out.println("Dog running");
+ 
+     return true;   
+ 
+ }  
+ 
+}
+ 
+```
+ 
+To create a new dog object.
+ 
+```java
+ 
+public class Dog_Main {
+ 
+public static void main(String[] args) {
+ 
+    Dog littleDog = new Dog("Bingo", 6,"Brown", "German Shepherd");
+ 
+    System.out.println("Is the dog running:? "+ littleDog.isrunning());
+ 
+    System.out.println();
+ 
+    System.out.println("is the dog sleeping:? " + littleDog.isSleeping());
+ 
+    System.out.println();
+ 
+    System.out.println("Is the dog barking:? " + littleDog.isbarking());
+ 
+    System.out.println();
+ 
+    System.out.println("Is the dog eating:? " + littleDog.iseating());
+ 
+    System.out.println();
+ 
+    System.out.println("The dog name is:  " + littleDog.name);
+ 
+}
+ 
+}
+ 
+```
+ 
+Now you see, we’ve created a new dog with the name `Bingo`. You can go on to create as many dogs with different states or attributes.
+ 
+Here is the output:
+ 
+  ```
+  Dog running
+ 
+  Is the dog running:? true
+ 
+  Sleeping
+ 
+  is the dog sleeping:? true
+ 
+  Dog barking...
+ 
+  Is the dog barking:? true
+ 
+  Eating
+ 
+  Is the dog eating:? true
+ 
+  The dog name is: Bingo
+  ```
+ 
+Now that we’ve got this out of the way let's talk about abstraction and encapsulation.
+ 
+### Abstraction 
+Who could have imagined that abstraction will be a thing in programming? Who? But right now, you can’t talk about OOP without mentioning abstraction. It is a significant building block of OOP. So then, what is abstraction? This section will explain what abstraction is and how to use it as a Java developer.
+
+In abstraction, only the useful attributes of an object are on display or accessible. The low-level details are *hidden*. For example, imagine you’re typing on your computer. You’re interested in punching the keys believing that it produces results on your screen. You know nothing about the mechanism that produces the result, and I’m sure you haven’t even thought about it yet. Have you?
+ 
+Here's another example to make things clearer: imagine car drivers. They are only interested in starting, moving, and stopping the car, speed, or changing gear and brake. They are not interested in the mechanism that allows them to do this. In abstraction, we hide the implementation details from the user while exposing the functionality.
+ 
+This way, the user will only know what a function (method in java) does but not how it does it.
+ 
+### How to implement Abstraction in Java
+In Java, abstraction is implemented using either of these two ways:
+- By using the non-access modifier *abstract* keyword or
+- Interface
+ 
+#### Working with the Abstract Keyword
```

Comment:
In this section and the next one on interfaces, I think you should mention how the compiler forces you to override/implement the abstract methods, or else the program won't run. Those can be short, 1 sentence explanations though.

Reply:
Done